addons/sale_stock/models/replenishment_warhouse_select.py

```diff
@@ -0,0 +1,45 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class SaleStockReplenishmentWarehouseSelect(models.TransientModel):
+    _name = 'sale_stock.replenishment.warehouse.select'
+    _description = 'Replenishment Warehouse Select'
+    _rec_name = 'sale_order_line_id'
+
+    sale_order_line_id = fields.Many2one('sale.order.line')
+    product_id = fields.Many2one('product.product', related='sale_order_line_id.product_id')
+    qty_to_order = fields.Float(related='sale_order_line_id.product_uom_qty')
+
+    # TODO: here need to retrieve all warehouses?
+    route_ids = fields.One2many(related='sale_order_line_id.order_id.warehouse_id.resupply_route_ids')
```

Comment:
@amoyaux For this one (the wizard on the sale order line), we don't want to find the routes by looking on the warehouse resupply routes, but by checking in all the company warehouses, right?

Reply:
@rdeodoo Yes, you can just take all the warehouse in the same company

Reply:
Perfect, I greened the task in RD Dev, I'd be interested in having a first review, functional or technical, to be sure it's going in the right direction and that I did not miss anything big.
Apart from the small framework stuff I'd like to triple check (to have one XML action instead of 2, but it's not related to stock at the end), this is LGTM for first review.

Reply:
@amoyaux 
As discussed IRL right now, I updated the PR to correctly (I think?) retrieve the routes:
```py
        customer_location = self.sale_order_line_id.order_id.partner_shipping_id.property_stock_customer
        routes = self.env['stock.rule'].search(expression.AND([
            self.env['stock.rule']._check_company_domain(self.env.company),
            [('location_dest_id', '=', customer_location.id)],
        ])).route_id
```
See the [diff](https://github.com/odoo/odoo/compare/fbe7bb3817a445e78702c64f8f6855a0ef0963ff..0eeb6c37a82ed145020d4ee9b62be51dddf45e2b)